drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -275,164 +288,147 @@ impl ServerImpl {
         // take an interrupt on ROT_IRQ falling edge with timeout.
         if !self.wait_rot_irq(true, timeout) {
             ringbuf_entry!(Trace::RotNotReady);
-            return Err(MsgError::RotNotReady);
+            return Err(SprotError::RotNotReady);
         }
 
         // Read just the header.
         // Keep CSn asserted over the two reads.
         ringbuf_entry!(Trace::CSnAssert);
         self.spi
             .lock(CsState::Asserted)
-            .map_err(|_| MsgError::SpiServerError)?;
+            .map_err(|_| SprotError::SpiServerError)?;
         if PART1_DELAY != 0 {
             hl::sleep_for(PART1_DELAY);
         }
 
-        // We can fetch FIFO size number of bytes reliably.
-        // After that, a short delay and fetch the rest if there is
-        // a payload.
-        // Small messages will fit entirely in the RoT FIFO.
-        //
-        // We don't, but we could speculate that some RoT responses will
-        // be longer than ROT_FIFO_SIZE and get ROT_FIFO_SIZE
-        // instead of MIN_MSG_SIZE.
-        //
-        // TODO: Use DMA on RoT to avoid this dance.
+        // Fill in rx_buf with a complete message and validate its crc
+        let res = self.do_read_response();
+
+        // We must release the SPI bus before we return
+        ringbuf_entry!(Trace::CSnDeassert);
+        self.spi.release().map_err(|_| SprotError::SpiServerError)?;
```

Comment:
Not required for this PR, but I sure would like to see an RAII-ish handle for the SPI bus...

Reply:
Totally agreed. Stoltz and I already discussed that before I let this PR fly. It looks like Cliff anticipated this with `pub fn lock_auto(&self, device_index: u8, assert_cs) -> Result<ControllerLock<'_>, SpiError> {}`. I just didn't notice it because I didn't look at the spi code at all.